src/engine.rs

```diff
@@ -177,13 +177,31 @@ pub fn run_treefmt(
 
             // Don't run the formatter if there are no paths to format!
             if paths.is_empty() {
-                (formatter_name.clone(), path_mtime.clone())
+                Ok((formatter_name.clone(), path_mtime.clone()))
             } else {
                 let start_time = Instant::now();
 
                 match formatter.clone().fmt(&paths) {
                     // FIXME: do we care about the output?
-                    Ok(_) => {
+                    Ok(out) => {
+                        if !out.status.success() {
+                            match out.status.code() {
+                                Some(scode) => {
+                                    return Err(anyhow!(
+                                        "{}'s formatter failed: exit status {}",
+                                        &formatter,
+                                        scode
+                                    ));
+                                }
+                                None => {
+                                    return Err(anyhow!(
+                                        "{}'s formatter failed: unknown formatter error",
+                                        &formatter
+                                    ));
+                                }
+                            }
+                        }
```

Comment:
Instead of exposing the Output, this should be handled in the Formatter fmt function. Then change the return signature of the function to `Result<()>`.